### PR TITLE
feat: swap over to new actions

### PR DESCRIPTION
**Pull request by kneal**

### CODE REVIEW

File: cmd/vela-terraform/main.go [hunk 1]
```diff
@@ -38,77 +38,108 @@ func main() {
...
+			Usage:   "the action to have terraform perform",
+			EnvVars: []string{"PARAMETER_ACTION"},
 		},
-		&cli.StringFlag{
-			Name:    "ca_cert",
-			Usage:   "ca cert to add to your environment to allow terraform to use internal/private resources",
-			EnvVars: []string{"PARAMETER_CA_CERT"},
+		&cli.BoolFlag{
+			Name:    "auto-approve",
+			Usage:   "an interactive approval of of running command",
```
Comment (wass3r) [inline cmd/vela-terraform/main.go:48]:
```suggestion
			Usage:   "skip interactive approval of running command",
```

File: cmd/vela-terraform/main.go [hunk 2]
```diff
@@ -38,77 +38,108 @@ func main() {
...
+			EnvVars: []string{"PARAMETER_CHECK"},
+		},
+		&cli.BoolFlag{
+			Name:    "check-variables",
+			Usage:   "command will check whether all required variables have been specified",
+			EnvVars: []string{"PARAMETER_CHECK_VARIABLES"},
+		},
+		&cli.BoolFlag{
+			Name:    "destroy",
+			Usage:   "will be generated to destroy all resources managed by the given configuration and state",
```
Comment (wass3r) [inline cmd/vela-terraform/main.go:68]:
```suggestion
			Usage:   "destroy all resources managed by the given configuration and state",
```

File: cmd/vela-terraform/main.go [hunk 3]
```diff
@@ -38,77 +38,108 @@ func main() {
...
+		},
+		&cli.BoolFlag{
+			Name:    "input",
+			Usage:   "ask for input for variables if not directly set",
+			EnvVars: []string{"PARAMETER_INPUT"},
+		},
+
+		&cli.BoolFlag{
+			Name:    "lock",
+			Usage:   "the state file when locking is supported",
```
Comment (wass3r) [inline cmd/vela-terraform/main.go:94]:
```suggestion
			Usage:   "lock the state file when locking is supported",
```

File: cmd/vela-terraform/main.go [hunk 4]
```diff
@@ -38,77 +38,108 @@ func main() {
...
+			Usage:   "specifies the depth of modules to show in the output",
+			EnvVars: []string{"PARAMETER_MODULE_DEPTH"},
 		},
-		&cli.StringFlag{
-			Name:    "netrc.machine",
-			Usage:   "netrc machine",
-			EnvVars: []string{"VELA_NETRC_MACHINE"},
+		&cli.BoolFlag{
+			Name:    "no-color",
+			Usage:   "the state file when locking is supported",
```
Comment (wass3r) [inline cmd/vela-terraform/main.go:109]:
```suggestion
			Usage:   "disables colors in output",
```

File: cmd/vela-terraform/main.go [hunk 5]
```diff
@@ -38,77 +38,108 @@ func main() {
...
+			Usage:   "number of concurrent operations as Terraform walks its graph",
+			EnvVars: []string{"PARAMETER_PARALLELISM"},
 		},
-		&cli.StringFlag{
-			Name:    "role_arn_to_assume",
-			Usage:   "A role to assume before running the terraform commands",
-			EnvVars: []string{"PARAMETER_ROLE_ARN_TO_ASSUME"},
+		&cli.BoolFlag{
+			Name:    "refresh",
+			Usage:   "update state prior to checking for differences/cal",
```
Comment (wass3r) [inline cmd/vela-terraform/main.go:124]:
```suggestion
			Usage:   "update state prior to checking for differences",
```

File: cmd/vela-terraform/main.go [hunk 6]
```diff
@@ -38,77 +38,108 @@ func main() {
...
+			Name:    "state",
+			Usage:   "path to read and save state",
+			EnvVars: []string{"PARAMETER_STATE"},
 		},
 		&cli.StringFlag{
-			Name:    "secrets",
-			Usage:   "a map of secrets to pass to the Terraform `plan` and `apply` commands. Each value is passed as a `<key>=<ENV>` option",
-			EnvVars: []string{"PARAMETER_SECRETS"},
+			Name:    "state-out",
+			Usage:   "path to write state to that is different than state",
```
Comment (wass3r) [inline cmd/vela-terraform/main.go:134]:
```suggestion
			Usage:   "path to write updated state file",
```

File: Dockerfile [hunk 7]
```diff
@@ -2,14 +2,18 @@
...
+ARG TERRAFORM_VERSION=0.12.26
 
 ################################################################################
 ##     docker build --no-cache --target binary -t vela-terraform:binary .     ##
 ################################################################################
 
 FROM alpine:latest as binary
 
-ENV TF_VERSION=0.12.24
+ARG TF_VERSION
```
Comment (wass3r) [inline Dockerfile:16]: do we want to set this to `TERRAFORM_VERSION`?

Comment (kneal) [reply Dockerfile:16]: Yes 😞

File: Dockerfile [hunk 8]
```diff
@@ -2,14 +2,18 @@
...
 ################################################################################
 ##     docker build --no-cache --target binary -t vela-terraform:binary .     ##
 ################################################################################
 
 FROM alpine:latest as binary
 
-ENV TF_VERSION=0.12.24
+ARG TERRAFORM_VERSION
 
 RUN wget -q https://releases.hashicorp.com/terraform/${TF_VERSION}/terraform_${TF_VERSION}_linux_amd64.zip -O terraform.zip && \
```
Comment (wass3r) [inline Dockerfile:18]: you will want to change `${TF_VERSION}` to `${TERRAFORM_VERSION}` now

Comment (kneal) [reply Dockerfile:18]: Didn't even look at that, only looked at the line you commented on 😞

Comment (wass3r) [reply Dockerfile]: my fault. should have called that out as well.

Comment (kneal) [reply Dockerfile]: I looked at the change without ☕ so tunnel vision sunk in 😬

**Pull request merged**